src/System.Private.CoreLib/shared/System/Threading/SemaphoreSlim.cs

```diff
@@ -349,7 +351,7 @@ public bool Wait(int millisecondsTimeout, CancellationToken cancellationToken)
                 try { }
```

Comment:
Is there a reason for the empty `try` block?

Reply:
Thread aborts by default are delayed over finally blocks, so this pattern is a way of creating abort-uninterruptable regions. That's not relevant to Core though; this was written this way originally for netfx.

Reply:
>so this pattern is a way of creating abort-uninterruptable regions

Did not know that. Is that an ECMA-335 thing or an implementation detail of the .NET CLR?

Reply:
I believe it's specific to .NET Framework; IIRC, ECMA 335 states that behavior related to asynchronous exceptions in general is implementation-specific.